spec/function.dd

```diff
@@ -2181,15 +2181,49 @@ void main()
 $(H2 $(LNAME2 interpretation, Compile Time Function Execution (CTFE)))
 
     $(P Functions which are both portable and free of global side-effects can be
-    executed at compile time. This is useful when constant folding
-    algorithms need to include recursion and looping. Compile time function
-    execution is subject to the following restrictions:
+    executed at compile time. In certain contexts, such compile time execution
+    is guaranteed. It is called Compile Time Function Execution (CTFE) then.
+    The contexts that trigger CTFE are:)
+
+    $(UL
+    $(LI initialization of a static variable or a manifest constant)
+    $(LI static initializers of struct/class members)
+    $(LI dimension of a static array)
+    $(LI argument for a template value parameter)
+    $(LI $(D_KEYWORD static if))
```

Comment:
I would recommend using `$(GLINK ...)` to link to the corresponding grammar, as that gives more information to the reader.

Reply:
In general, I think the more the better, but of course in reasonable manner. E.g. search for `GLINK` in some of Walter's recent PRs to the spec (https://github.com/dlang/dlang.org/pull/2113/files, https://github.com/dlang/dlang.org/pull/2104/files).